lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -40,49 +40,47 @@ const rule = (primary, secondaryOptions) => {
 			return;
 		}
 
-		const longhandProperties = Object.entries(shorthandData).reduce(
-			(/** @type {Record<string, string[]>} */ longhandProps, [key, values]) => {
-				if (optionsMatches(secondaryOptions, 'ignoreShorthands', key)) {
-					return longhandProps;
-				}
+		/** @type {Map<string, string[]>} */
+		const longhandToShorthands = new Map();
 
-				for (const value of values) {
-					(longhandProps[value] || (longhandProps[value] = [])).push(key);
-				}
+		for (const [shorthand, longhandProps] of longhandSubPropertiesOfShorthandProperties.entries()) {
+			if (optionsMatches(secondaryOptions, 'ignoreShorthands', shorthand)) {
+				continue;
+			}
 
-				return longhandProps;
-			},
-			{},
-		);
+			for (const longhand of longhandProps) {
+				const shorthands = longhandToShorthands.get(longhand) || [];
+
+				shorthands.push(shorthand);
+				longhandToShorthands.set(longhand, shorthands);
+			}
+		}
```

Comment:
[note] Refactor to use `Map` and `for...of` instead of `Object` and `reduce()` for readability.